### PR TITLE
Add unit tests for burp.Methords

### PR DESCRIPTION
**Pull request by ThomasPerkins1123**
Hi,

I've analysed your code base and noticed that `burp.Methods` in the `knife` module is not fully tested.

I've written some tests that cover this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -86,6 +86,26 @@
 			<version>1.6</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.powermock</groupId>
+			<artifactId>powermock-api-mockito</artifactId>
+			<version>1.6.5</version>
```
Comment (Braavos6) [inline pom.xml]: Try to keep the versions the same.. Use 1.6.6 for both or 1.6.5

File: pom.xml [hunk 2]
```diff
@@ -86,6 +86,26 @@
...
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>4.12</version>
+			<scope>compile</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.powermock</groupId>
+			<artifactId>powermock-module-junit4</artifactId>
+			<version>1.6.6</version>
```
Comment (Braavos6) [inline pom.xml:105]: Try to keep the versions the same.. Use 1.6.6 for both or 1.6.5

File: src/test/burp/MethodsTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            Methods.getStrList("fooBar123", 3, 2));
+  }
+
+  @Test
+  public void testSubstring() {
+    Assert.assertEquals("345",
+            Methods.substring("12345", 2, 5));
+    Assert.assertEquals("45",
+            Methods.substring("12345", 3, 6));
+    Assert.assertNull(Methods.substring("12345", 6, 7));
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: Add a blank line between different types of Asserts. Put assertNull or assertNotNull first and then blank line then other asserts

File: src/test/burp/MethodsTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @PrepareForTest({JOptionPane.class, Methods.class})
+  @Test
+  public void testPrompt_and_validate_input() {
+    PowerMockito.mockStatic(JOptionPane.class);
+    PowerMockito.when(JOptionPane.showInputDialog(anyString(),
+            anyString())).thenReturn("").thenReturn("fooBar");
+    Assert.assertEquals("fooBar",
+            Methods.prompt_and_validate_input("foo", "Bar"));
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: Your missing a assert for the ` if (null == user_input) return  null;` in `prompt_and_validate_input()` method

Comment (Braavos6) [reply src/test/burp/MethodsTest.java]: Or I can't find it... Or these is something wrong with the method and I didn't read you PR text properly :)

File: src/test/burp/MethodsTest.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  @Test
+  public void testDecimalToHex() {
+    Assert.assertEquals("7C", Methods.decimalToHex(124));
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java:41]: I would add a test case for decimal value `0`

File: src/test/burp/MethodsTest.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            Methods.do_modify_request("foo".getBytes(),
+                    new int[]{3, 2}, null));
+  }
+
+  @PrepareForTest({Random.class, Methods.class})
+  @Test
+  public void testGetRandomString() throws Exception {
+    final Random random = PowerMockito.mock(Random.class);
+    PowerMockito.when(random.nextInt(anyInt()))
+            .thenReturn(5).thenReturn(14).thenReturn(14);
```
Comment (Braavos6) [inline src/test/burp/MethodsTest.java]: I maybe wrong here but I think you can do this with one `.thenReturn()`

**Review by Braavos6 — CHANGES_REQUESTED** [review]

File: src/test/burp/MethodsTest.java (hunk 4 above)
Comment (ThomasPerkins1123) [reply src/test/burp/MethodsTest.java]: yes you are right cover skipped that one and I didn't spot that

File: src/test/burp/MethodsTest.java (hunk 6 above)
Comment (ThomasPerkins1123) [reply src/test/burp/MethodsTest.java]: Didn't realise you could do that. Was just the format that cover would produce and i had never actually questioned it

Comment (ThomasPerkins1123) [reply src/test/burp/MethodsTest.java]: also the extra 14 on the end was redundant so can be shortened to `.thenReturn(5, 14)`

File: src/test/burp/MethodsTest.java (hunk 3 above)
Comment (Braavos6) [reply src/test/burp/MethodsTest.java]: @ThomasPerkins1123 Flip the Asserts around.. So the AssertNull and NotNull come first then blank line then other Asserts :)

**Comment by Braavos6** [comment]
Remove the other files being committed and only keep the 3 files

**Comment by ThomasPerkins1123** [comment]
> Remove the other files being committed and only keep the 3 files

The other files are just being moved as the directory structure is being changed to allow for adding the unit tests.
See the other PRs for more info

**Review by Braavos6 — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Review by freddytuxworth — APPROVED** [review]

**Pull request closed without merging**